framework/src/org/checkerframework/framework/util/typeinference/DefaultTypeArgumentInference.java

```diff
@@ -635,8 +640,13 @@ private void handleUninferredTypeVariables(
                 final AnnotatedTypeMirror inferredType = inferredArgs.get(typeVar);
 
                 if (inferredType == null) {
-                    AnnotatedTypeMirror dummy = typeFactory.getUninferredWildcardType(atv);
-                    inferredArgs.put(atv.getUnderlyingType(), dummy);
+                    AnnotatedTypeMirror dummy = atv.getUpperBound().deepCopy();
```

Comment:
I don't understand what seems wrong about using the upper bound.  Previously, we created a new wildcard and set its upper bound to the upper bound of the type var.  The upper bound of that type variable might also have been a type variable.